src/include/simeng/models/inorder/Core.hh

```diff
@@ -140,28 +139,50 @@ class Core : public simeng::Core {
   pipeline::PipelineBuffer<MacroOp> fetchToDecodeBuffer_;
 
   /** The buffer between decode and execute. */
-  pipeline::PipelineBuffer<std::shared_ptr<Instruction>> decodeToExecuteBuffer_;
+  pipeline::PipelineBuffer<std::shared_ptr<Instruction>> decodeToIssueBuffer_;
+
+  /** The issue ports; single-width buffers between issue and execute. */
+  std::vector<pipeline::PipelineBuffer<std::shared_ptr<Instruction>>>
+      issuePorts_;
 
   /** The buffer between execute and writeback. */
   std::vector<pipeline::PipelineBuffer<std::shared_ptr<Instruction>>>
       completionSlots_;
 
-  /** The previously generated addresses. */
-  std::queue<simeng::memory::MemoryAccessTarget> previousAddresses_;
+  /** The core's load/store queue. */
+  pipeline::LoadStoreQueue loadStoreQueue_;
 
   /** The fetch unit; fetches instructions from memory. */
   pipeline::FetchUnit fetchUnit_;
 
   /** The decode unit; decodes instructions into uops and reads operands. */
   pipeline::DecodeUnit decodeUnit_;
 
-  /** The execute unit; executes uops and sends to writeback, also forwarding
-   * results. */
-  pipeline::ExecuteUnit executeUnit_;
+  /** The inorder stager unit; tracks in program-order instructions issued to
+   * ensure writeback functionality occurs in program-order. */
+  pipeline::InOrderStager staging_;
+
+  /** The issue unit; reads operands, and issues ready instructions to the
+   * execution unit in program-order. */
+  pipeline::BlockingIssueUnit issueUnit_;
+
+  /** The set of execution units; executes uops and sends to writeback, also
+   * forwarding results to issue. */
```

Comment:
Would it be better if the comment said: "...also forwarding results to the issue unit or blocking issue unit"

Reply:
The comment doesn't gain much from the extra specifics suggested. It's just less concise. Results forwarding is a well-known/defined process so the comment's conciseness works here.